fairlearn/reductions/_moments/conditional_selection_rate.py

```diff
@@ -67,7 +157,26 @@ def gamma(self, predictor):
 
     # TODO: this can be further improved using the overcompleteness in group membership
     def project_lambda(self, lambda_vec):
-        """Return the projected lambda values."""
+        r"""Return the rebalanced Lagrange multipliers.
+
+        For each constraint, we have two Lagrange multipliers corresponding to the
+        positive and negative violations of the constraint, with both multipliers
+        constrained to be positive. The 'true' Lagrange multiplier is then
+        :math:`\lambda = \lambda_{+} - \lambda_{-}`. Since it makes little
+        sense to the positive and negative constraints violated at the same time,
+        this routine rebalances between them. For example, if we have
+        :math:`\lambda_{+}=3` and :math:`\lambda_{-}=1` then this method will
+        make :math:`\lambda_{+}=2` and :math:`\lambda_{-}=0`. Similarly if
+        :math:`\lambda_{+}=2` and :math:`\lambda_{-}=3` then this method will
+        yield :math:`\lambda_{+}=0` and :math:`\lambda_{-}=1`.
+
+        The supplied :class:`pandas:pandas.DataFrame` must have the same
+        :class:`pandas:pandas.MultiIndex` as `index` and the resultant
+        :class:`pandas:pandas.DataFrame` will have that `index` as well.
+
+        :param lambda_vec: The set of Lagrange multipliers indexed by `index`
+        :type lambda_vec: :class:`pandas:pandas.DataFrame`
```

Comment:
It's a little odd that (assuming I didn't mess up my testing) this method requires `lambda_vec` to be a DataFrame, while the next one requires a Series.